### ISSUE TITLE
Doc update needed for pyjnius&threads & other points

### ISSUE THREAD

**Issue by pakal** [kivy/pyjnius#469]
I read about jnius.detach() in the docs, and began investigating how to do it in my, when I stumbled upon a hack in jnius/__init__.py which automatically calls jnius.detach() for native python threads on Android. That's why I never had any crash with that :).  May I update the docs to notify this android-specific case? 

Also, maybe I should add a note about the special "android.os.Build$VERSION" notation, which non-java coders like me didn't get at all initially?

Maybe also indicate that an Android, jnius loads the existing JVM (the one which started the p4a app), not a new one, that's why autoclass("org.kivy.android.PythonActivity").mActivity returns a proper value?

I'll make a PR according to what's suitable.

**Comment by tshirtman** [comment]
Yes, improvements to the doc are welcome. https://pyjnius.readthedocs.io/en/stable/android.html the android section already has a few specific indications, but it's certainly worth mentioning the hack to make life easier on android.
The notation for class attributes (i think that's how they are called, but I'm not a java dev either), is indeed used but not explained at all in the doc, it could certainly help newcomers a bit to have at least a little blurp about them.
Nothing against the making clearer that you are accessing the jvm the application is running into in android, although that did seem obvious to me, as it would be much harder to do anything useful without that, but nothing wrong with making it explicit.

**Comment by pakal** [comment]
Ok right I'll have a look at it.

On a side note, I saw it's the run() method of Thread which is monkey-patched, but if users subclass Thread and implement their own run(), the jni.detach() is lost (it should have happened to me, i dunno why the dalvik VM didn't crash).
Wouldn't it be possible to monkey-patch the "_bootstrap_inner" of Thread instead, so that jni.detach() is called in any case?  (else I'll mention this gotcha in the docs)

**Comment by tshirtman** [comment]
I actually don't know if this would cause a problem, it's worth investigating, i didn't do this workaround myself, and i don't know if a lot of thoughts went into how to make it the most proper, it's entirely possible it was done as a quick fix and has a lot of room for improvements, if you want to look into it, feel free, maybe we should open an issue about improving this part.

**Comment by pakal** [comment]
Here is a doc update, the run() monkey-patching should do the work for 99% cases so I guess it could live as is  B-)

https://github.com/kivy/pyjnius/pull/478

**Comment by tshirtman** [comment]
Merged,  thanks!

**Comment by pakal** [comment]
thanks!
